lib/rubocop/cop/style/object_methods.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop enforces the use of consistent method names
+      # from the Object class.
+      #
+      # You can customize the mapping from undesired method to desired method.
+      #
+      # e.g. to use `yield_self` over `then`:
+      #
+      #   Style/CollectionMethods:
+      #     PreferredMethods:
+      #       then: yield_self
+      #
+      # @example
+      #   # These examples are based on the default mapping for `PreferredMethods`.
+      #
+      #   # bad
+      #   obj.yield_self
+      #
+      #   # good
+      #   obj.then
```

Comment:
I see cases where block arguments are used.

```suggestion
      #   # bad
      #   obj.yield_self { |x| x.do_something }
      #
      #   # good
      #   obj.then { |x| x.do_something }
```

Reply:
Oops, Sorry. I’ll add it.